core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -71,7 +71,7 @@ public static function getTemplate($strTemplate)
 			/** @var PageModel $objPage */
 			global $objPage;
 
-			if ($objPage->templateGroup)
+			if ($objPage && $objPage->templateGroup)
```

Comment:
This change is where I am most uncertain. Can `$objPage` legally be `null` here or is this a problem with my setup?

Reply:
When does this occur in your setup?

Reply:
I saw it in my log, but it did not give any obvious context. I’ll try to reproduce this, please stand by.

Reply:
From studying the logs, it happens while the daily cron jobs are executed. `$strTemplate` is set to `mod_article`. Maybe that rings a bell for someone? Otherwise, I’ll add more instrumentation.

Reply:
Do you know which cronjob causes that? You can also look at `tl_cron_job` to see which cronjobs are getting executed in your system and when.

Reply:
I looked in that table and when the log messages appeared, the minutely, hourly and daily jobs executed (the weekly or monthly ones did *not* run). I then re-ran the minutely and hourly ones and nothing happened in the log. I therefore suspect the daily jobs.

Reply:
In theory this should never happen though. Even if a cron job renderes some templates (for instance `News::generateFeeds` for the content elements of a news entry), `TL_MODE` shouldn't be defined there. You would need to debug which cron job causes this and  figure out why `TL_MODE` is set to `FE`.

Reply:
I see, thank you for debugging :).

@contao/developers I also not sure what the best course of action is here. On the one hand I think we should replace these instances of

```php
if (\defined('TL_MODE') && TL_MODE == 'FE')
```

with

```php
$request = System::getContainer()->get('request_stack')->getCurrentRequest();

if ($request && System::getContainer()->get('contao.routing.scope_matcher')->isFrontendRequest($request))
```

similar to what we already did in https://github.com/contao/contao/pull/2628. However, an additional check for `null !== $objPage` would certainly not hurt too - but I am not sure this would really be the best way.

Reply:
How would replacing the instances fix the issue?

Reply:
I also don’t see the connection. From my very cursory understanding of Contao’s internals, if `getArticle` is invoked because on an insert tag, it is somewhat correct that `$objPage` is `null`.